plugins/home/src/api/VisitsWebStorageApi.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ErrorApi, IdentityApi } from '@backstage/core-plugin-api';
+import { VisitsStorageApi } from './VisitsStorageApi';
+import { WebStorage } from '@backstage/core-app-api';
+
+/** @public */
+export type VisitsWebStorageApiOptions = {
+  limit?: number;
+  identityApi: IdentityApi;
+  errorApi: ErrorApi;
+};
+
+/**
+ * @public
+ * This is a reference implementation of VisitsApi using WebStorage.
+ */
+export class VisitsWebStorageApi {
+  static create(options: VisitsWebStorageApiOptions) {
+    return VisitsStorageApi.create({
+      limit: options.limit,
+      identityApi: options.identityApi,
+      storageApi: WebStorage.create({ errorApi: options.errorApi }),
```

Comment:
Question (non-blocking):
This implementation seems to only exist to provide a web storage implementation for the visits storage API. Why not provide it as the default implementation of the storage API while creating the homepage plugin plugin?

Reply:
Although I like the idea of providing a default implementation through the WebStorage, it would inject a new dependency. The WebStorage depends on an errorApi. You can provide your own implementation of the StorageApi that it is not dependent upon the errorApi that WebStorage is.

Reply:
The `StorageApi` would still not depend on the `ErrorApi`, the ErrorApi` is a dependency of the `WebStorage` implementation only... Also, the adopter will likely have to provide an `IdentityApi` anyway, they can also provide an Error API if they want to use WebStorageApi as the default API.

Reply:
I confess I don't understand you. Can you come up with a patch? Maybe it would be clearer in code.

Reply:
Thanks @camilaibs! After our chat I've tested this idea. The storageApi is already provided as a default api so I've kept only the visitsApi in the plugin.ts. But I've followed your idea and changed the usage on the development app where instead of configuring the visitsApi in api.ts I've added the homePlugin in plugins.ts (all in the packages/app).